src/test/java/io/redis/examples/async/HashExample.java

```diff
@@ -0,0 +1,166 @@
+                System.out.println(res6); // >>> 5072
+                // REMOVE_START
+                assertThat(res6).isEqualTo(5072L);
+                // REMOVE_END
+                return asyncCommands.hincrby("bike:1", "price", -100);
+            })
+                    // REMOVE_START
+                    .thenApply(res -> {
+                        assertThat(res).isEqualTo(4972L);
+                        return res;
+                    })
+                    // REMOVE_END
+                    .thenAccept(System.out::println)
+                    // >>> 4972
+                    .toCompletableFuture();
+            // STEP_END
+
+            // STEP_START incrby_get_mget
+            CompletableFuture<Void> incrByGetMget = asyncCommands.hincrby("bike:1:stats", "rides", 1).thenCompose(res7 -> {
+                System.out.println(res7); // >>> 1
+                // REMOVE_START
+                assertThat(res7).isEqualTo(1L);
+                // REMOVE_END
+                return asyncCommands.hincrby("bike:1:stats", "rides", 1);
+            }).thenCompose(res8 -> {
+                System.out.println(res8); // >>> 2
+                // REMOVE_START
+                assertThat(res8).isEqualTo(2L);
+                // REMOVE_END
+                return asyncCommands.hincrby("bike:1:stats", "rides", 1);
+            }).thenCompose(res9 -> {
+                System.out.println(res9); // >>> 3
+                // REMOVE_START
+                assertThat(res9).isEqualTo(3L);
+                // REMOVE_END
+                return asyncCommands.hincrby("bike:1:stats", "crashes", 1);
+            }).thenCompose(res10 -> {
+                System.out.println(res10); // >>> 1
+                // REMOVE_START
+                assertThat(res10).isEqualTo(1L);
+                // REMOVE_END
+                return asyncCommands.hincrby("bike:1:stats", "owners", 1);
+            }).thenCompose(res11 -> {
+                System.out.println(res11); // >>> 1
+                // REMOVE_START
+                assertThat(res11).isEqualTo(1L);
+                // REMOVE_END
+                return asyncCommands.hget("bike:1:stats", "rides");
+            }).thenCompose(res12 -> {
+                System.out.println(res12); // >>> 3
+                // REMOVE_START
+                assertThat(res12).isEqualTo("3");
+                // REMOVE_END
+                return asyncCommands.hmget("bike:1:stats", "crashes", "owners");
+            })
+                    // REMOVE_START
+                    .thenApply(res -> {
+                        assertThat(res.toString()).isEqualTo("[KeyValue[crashes, 1], KeyValue[owners, 1]]");
+                        return res;
+                    })
+                    // REMOVE_END
+                    .thenAccept(System.out::println)
+                    // >>> [KeyValue[crashes, 1], KeyValue[owners, 1]]
+                    .toCompletableFuture();
+            // STEP_END
+
+            CompletableFuture.allOf(
```

Comment:
My bad, deleted my previous comment.

Why only add `hIncrBy` and `incrByGetMget` ? Perhaps we need to add all the completable futures??

Reply:
My bad, they are all used as basis for the other completable futures, thanks for clarifying offline.